Source/MQTTnet/Client/MqttClient.cs

```diff
@@ -751,10 +755,24 @@ async Task<MqttClientPublishResult> PublishAtLeastOnce(MqttPublishPacket publish
 
         async Task<MqttClientPublishResult> PublishAtMostOnce(MqttPublishPacket publishPacket, CancellationToken cancellationToken)
         {
-            // No packet identifier is used for QoS 0 [3.3.2.2 Packet Identifier]
-            await Send(publishPacket, cancellationToken).ConfigureAwait(false);
+            try
+            {
+                // No packet identifier is used for QoS 0 [3.3.2.2 Packet Identifier]
+                await Send(publishPacket, cancellationToken).ConfigureAwait(false);
+
+                return MqttClientResultFactory.PublishResult.Create(null);
+            }
+            catch (Exception)
+            {
+                MqttDisconnectPacket localUnexpectedDisconnectPacket = _unexpectedDisconnectPacket;
 
-            return MqttClientResultFactory.PublishResult.Create(null);
+                if (localUnexpectedDisconnectPacket != null)
+                {
+                    throw new MqttClientUnexpectedDisconnectReceivedException(localUnexpectedDisconnectPacket);
```

Comment:
I would pass the actual exception to the inner exception so that it is still accessible for the user.